lib/lists.ex

```diff
@@ -0,0 +1,582 @@
+defmodule Lists do
+  @moduledoc """
+  See https://sites.google.com/site/prologsite/prolog-problems/1
+  """
+
+  @doc """
+  Find the last element in a list.
+
+    iex> Lists.last([:a, :b, :c, :d])
+    :d
+  """
+  def last([element]), do: element
+  def last([_ | t]), do: last(t)
+
+  @doc """
+  Find the last but one element in a list.
+
+    iex> Lists.last_but_one([:a, :b, :c, :d])
+    :c
+  """
+  def last_but_one([penultimate, _ultimate]), do: penultimate
+  def last_but_one([_ | t]), do: last_but_one(t)
+
+  @doc """
+  Find the K'th element of a list.
+
+  The first element in the list is number 1.
+
+    iex> Lists.element_at([:a, :b, :c, :d, :e], 3)
+    :c
+  """
+  def element_at([h | _], 1), do: h
+
+  def element_at([_, h1 | t], n) when n > 1 do
+    element_at([h1 | t], n - 1)
+  end
+
+  @doc """
+  Find the number of elements of a list.
+
+    iex> Lists.len([:a, :b, :c])
+    3
+  """
+  def len(my_list), do: len(my_list, 0)
+  defp len([], acc), do: acc
+  defp len([_ | t], acc), do: len(t, acc + 1)
+
+  @doc """
+  Reverse a list.
+
+    iex> Lists.reverse([:a, :b, :c])
+    [:c, :b, :a]
+  """
+  def reverse(my_list), do: reverse(my_list, [])
+  defp reverse([], acc), do: acc
+  defp reverse([h | t], acc), do: reverse(t, [h | acc])
+
+  @doc """
+  Find out whether a list is a palindrome.
+
+    iex> Lists.is_palindrome([:x, :a, :m, :a, :x])
+    true
+    iex> Lists.is_palindrome([:x, :a, :m, :a])
+    false
+  """
+  def is_palindrome([]), do: false
```

Comment:
It would be more elixiry to call it `palindrome?`

Reply:
although weirdly there are functions like `is_even` and `is_odd`

Reply:
Anything that can be used in a guard clause uses the `is_xxx` format for consistency with Erlang.

Pure Elixir functions that return a boolean are in the conventino `xxx?`

Reply:
Yeah, definitely. I only called it `is_palindrome` because I was trying to match each function to the one defined in the prolog sites.
But with elixir `palindrome?` makes more sense.